replicatelogs.go

```diff
@@ -677,3 +689,42 @@ func readTenantMassifChanges(ctx context.Context, cCtx *cli.Context, cmd *CmdCtx
 	// No explicit config and --all not set, read from stdin
 	return stdinToDecodedTenantMassifs()
 }
+
+// prefetchingSealReader pre-fetches the seal for the massif to avoid racing with the
+// sealer.  If the massif is read first, the log can grow and a a new seal can
+// be applied to the *longer* log. At which point the previously read copy of
+// the massif will be "to short" for the seal.
+// See Bug#10530
+type prefetchingSealReader struct {
```

Comment:
this feels a bit wrong to have a type here?

Why can't we just pass the seal around after we have acquired it before the massif?

Reply:
I don't agree. We already pass the sealGetter via the options. This change simply uses that existing mechanism to pass a "pre fetched seal". And that results in a much simpler interface for most uses. For example local reader is perfectly happy to fetch on demand in verifyContext.